.github/workflows/build_deploy_exm_be.yml

```diff
@@ -0,0 +1,103 @@
+name: Build and Deploy EXM BE
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  build:
+    name: Build Backend Application
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set up Java 17
+      uses: actions/setup-java@v2
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+
+    - name: Install Maven
+      run: sudo apt-get update && sudo apt-get install maven -y
+
+    - name: Build Backend Application
+      run: mvn clean install
+      working-directory: expenses-manager-backend
+
+    - name: Upload build artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: backend-build
+        path: expenses-manager-backend/target/*.jar
+
+  deploy:
+    name: Deploy to VM
+    runs-on: ubuntu-latest
+    needs: build
+    environment: deploy
+    if: github.event_name == 'push' || github.event_name == 'workflow_dispatch' || github.event_name == 'pull_request'
```

Comment:
Nu cred ca vrem sa facem deploy la pull request, doar build, deci merge scos event pull_request de aici, si vrem sa facem deploy doar la push in main sau cu workflow dispatch